src/resolvers/deck/createDeck.ts

```diff
@@ -90,90 +96,90 @@ export const importDeck = mutationWithClientMutationId({
-          helperMap.updateFieldIndex,
-          template.frontSide
-        )
-        const backSideWithUpdatedEntities = updateEntitiesIds(
-          helperMap.updateFieldIndex,
-          template.backSide
+        helperMap.updateModelIndex[model.model?._id] = newModel._id
+
+        // Create fields
+        await Promise.all(
+          model.fields.map(async (field) => {
+            const currentField = await FieldModel.create({
+              name: field.name,
+              modelId: newModel._id,
+            })
+            helperMap.updateFieldIndex[field._id] = currentField._id
+          })
         )
 
-        await TemplateModel.create({
-          name: template.name,
-          modelId: newModel._id,
-          ownerId: user?._id,
-          frontSide: frontSideWithUpdatedEntities,
-          backSide: backSideWithUpdatedEntities,
-        })
-      }
-    }
+        // Create templates
+        await Promise.all(
+          model.templates.map(async (template) => {
+            const frontSideWithUpdatedEntities = updateEntitiesIds(
+              helperMap.updateFieldIndex,
+              template.frontSide
+            )
+            const backSideWithUpdatedEntities = updateEntitiesIds(
+              helperMap.updateFieldIndex,
+              template.backSide
+            )
+
+            await TemplateModel.create({
+              name: template.name,
+              modelId: newModel._id,
+              ownerId: user?._id,
+              frontSide: frontSideWithUpdatedEntities,
+              backSide: backSideWithUpdatedEntities,
+            })
+          })
+        )
+      })
+    )
 
     // Create notes
-    for await (const note of deckNotes) {
-      const currentNoteModelId = helperMap.updateModelIndex.find(
-        (v) => `${v.oldId}` === `${note.modelId}`
-      )?.newId
+    await Promise.all(
+      deckNotes.map(async (note) => {
+        const currentNoteModelId =
+          helperMap.updateModelIndex[note.modelId.toString()]
+
+        const fieldValues = note.values.map((noteValue) => {
+          const newFieldId =
+            helperMap.updateFieldIndex[noteValue.fieldId.toString()]
+
+          return {
+            data: removeContentStateDocumentIds(noteValue.data),
+            fieldId: newFieldId,
+          }
+        })
 
-      const fieldValues = note.values.map((noteValue) => {
-        const newFieldId = helperMap.updateFieldIndex.find(
-          (v) => `${v.oldId}` === `${noteValue.fieldId}`
-        )?.newId
+        const newNote = await NoteModel.create({
+          modelId: currentNoteModelId,
+          deckId: newDeck._id,
+          ownerId: user!._id,
+          values: fieldValues,
+          flashCards: [],
+        })
 
-        return {
-          data: removeContentStateDocumentIds(noteValue.data),
-          fieldId: newFieldId,
-        }
-      })
+        // Create flashcards
+        const modelTemplates = await TemplateModel.find({
```

Comment:
this query will be executed for every note you need to create, but we really only need to query it once right? you can move this to line 142, or ideally you can store the templates we create in memory and avoid a query to mongodb altogether